cpp/Platform.Delegates/Platform.Delegates.MulticastDelegate.h

```diff
@@ -98,26 +60,26 @@ namespace Platform::Delegates
         {
             if (this != &other)
             {
-                MoveCallbacksUnsync(other);
+                MoveCallbacksUnsync(std::move(other));
```

Comment:
This is a bug? The std::move was required? What exactly it is doing here?

Reply:
This is a famous story, see an example:
```c++
void foo(A&& a) {
    std::cout << "rvalue\n";
}

void foo (const A& a) {
    std::cout << "lvalue\n";
}

void bar(A && a) {
    foo(a);
    foo(std::move(a));
}
```
Here will be printed first lvalue, then rvalue. If you need to call an rvalue function with an rvalue value, you need to use std::move anyway. And this is exactly the case, because you can move data from an object only if it is rvalue

Reply:
Perfect, thank you!